### DISCUSSION TITLE
API call to renew a license

### DISCUSSION THREAD

**Discussion by fmg-pete** [keygen-sh/community#68] [Q&A] [answered]
I have built an admin tool that is able to generate different license keys for our product, but I can't figure out how to do a license renewal. If I try and use an existing email address, I get an error letting me know that it's already been used. I then decided to try and look up users by email account, but after a few hours of different approaches, any API request I make either gets ALL users, or none.

I first tried this:

`https://api.keygen.sh/v1/accounts/{account_id}/users?filter[email]=email@iknowexistsinthere.com`

Then this:

`https://api.keygen.sh/v1/accounts/?filter[email]=email@iknowexistsinthere.com`

And then asking chatGPT :D

`<?php

$email = 'example@example.com';
$apiEndpoint = 'https://api.keygen.sh/v1/accounts?filter[email]=' . urlencode($email);

$curl = curl_init();

curl_setopt_array($curl, [
  CURLOPT_URL => $apiEndpoint,
  CURLOPT_RETURNTRANSFER => true,
  CURLOPT_HTTPHEADER => [
    'Authorization: Bearer YOUR_API_KEY',
    'Content-Type: application/vnd.api+json',
  ],
]);

$response = curl_exec($curl);
$err = curl_error($curl);

curl_close($curl);

if ($err) {
  // Handle errors
  echo "cURL Error #:" . $err;
} else {
  // Handle the response data
  $data = json_decode($response, true);
  var_dump($data);
}`

But so far, no luck. :(

Any ideas? Seems like I am missing something really obvious.

**Comment by ezekg** [comment, marked as answer]
You can use the user’s email in place of their ID:

```
GET /v1/accounts/<account>/users/email@iknowexistsinthere.com
```

**Comment by fmg-pete** [comment]
So quick! That was it! Thankyou so much. 🙌
